python/pyspark/sql/protobuf/functions.py

```diff
@@ -0,0 +1,196 @@
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+A collections of builtin protobuf functions
+"""
+
+
+from typing import Dict, Optional, TYPE_CHECKING
+from pyspark import SparkContext
+from pyspark.sql.column import Column, _to_java_column
+from pyspark.util import _print_missing_jar
+
+if TYPE_CHECKING:
+    from pyspark.sql._typing import ColumnOrName
+
+
+def from_protobuf(
+    data: "ColumnOrName",
+    descFilePath: str,
+    messageName: str,
+    options: Optional[Dict[str, str]] = None,
+) -> Column:
+    """
+    Converts a binary column of Protobuf format into its corresponding catalyst value.
+    The specified schema must match the read data, otherwise the behavior is undefined:
+    it may fail or return arbitrary result.
+    To deserialize the data with a compatible and evolved schema, the expected
+    Protobuf schema can be set via the option protobuf descriptor.
+
+    .. versionadded:: 3.4.0
+
+    Parameters
+    ----------
+    data : :class:`~pyspark.sql.Column` or str
+        the binary column.
+    descFilePath : str
+        the protobuf descriptor in Message GeneratedMessageV3 format.
+    messageName: str
+        the protobuf message name to look for in descriptorFile.
+    options : dict, optional
+        options to control how the Protobuf record is parsed.
+
+    Notes
+    -----
+    Protobuf functionality is provided as an pluggable external module since Spark 3.4.0.
+
+    Examples
+    --------
+    >>> from pyspark.sql import Row
+    >>> import pyspark.sql.types as T
+    >>> from pyspark.sql.protobuf.functions import from_protobuf, to_protobuf
+    >>> data = ([Row(key="1", value=Row(age=2, name="Alice", score=109200))])
+    >>> ddl_schema = "key string, value struct<age: integer, name: string, score: long>"
+    >>> df = spark.createDataFrame(data, T._parse_datatype_string(ddl_schema))
+    >>> descFilePath = 'connector/protobuf/src/test/resources/protobuf/pyspark_test.desc'
+    >>> messageName = 'SimpleMessage'
+    >>> protobufDf = df.select(to_protobuf(df.value, descFilePath, messageName).alias("protobuf"))
+    >>> protobufDf.show(truncate=False)
+    +----------------------------------------+
+    |protobuf                                |
+    +----------------------------------------+
+    |[08 02 12 05 41 6C 69 63 65 18 90 D5 06]|
+    +----------------------------------------+
+
+    >>> descFilePath = 'connector/protobuf/src/test/resources/protobuf/pyspark_test.desc'
+    >>> messageName = 'SimpleMessage'
+    >>> df = protobufDf.select(from_protobuf(protobufDf.protobuf, \
+    descFilePath, messageName).alias("value"))
```

Comment:
```suggestion
    >>> df = protobufDf.select(
    ...     from_protobuf(protobufDf.protobuf, descFilePath, messageName).alias("value"))
```

Using slashes are actually discouraged according to PEP8 IIRC.